### ISSUE TITLE
Deprecation of package ioutil in Go 1.16

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#1019]
With the two proposals accepted (https://github.com/golang/go/issues/42026 and https://github.com/golang/go/issues/40025), the package `io/ioutil` will be deprecated and new code is encouraged to use the respective implementations in the packages `io` and `os`.

For example:

```Go
ioutil.ReadFile(...)
```

would become

```Go
os.ReadFile(...)
```

or

```Go
ioutil.ReadAll(...)
```

would become

```Go
io.ReadAll(...)
```

**Comment by cristaloleg** [comment]
Looks like those cases can be handled by `ruleguard` 👀 

Also, if those (all) declarations from `ioutil` are marked with a `Deprecated: ` phrase in doc-comment, `deprecatedComment` checker should catch this (nice to check this!)

**Comment by quasilyte** [comment]
For the ruleguard we would need a target Go version tag argument or inference.

The same goes for go-critic.
But since this checker can be easily disabled, we can avoid the need for the feature flag.
If we'll ever implement it on top of ruleguard rules, nothing would change for the gocritic users.
So, in the end, I think it's OK to implement it in gocritic.

> Also, if those (all) declarations from ioutil are marked with a Deprecated: phrase in doc-comment, deprecatedComment checker should catch this (nice to check this!)

Good point.
But it looks that they're not marked with `Deprecated:`.

https://tip.golang.org/src/io/ioutil/ioutil.go?s=1167:1213#L26

Maybe we can send a CL to mark them as such? I wonder if there is a reason behind them not being marked in the first place.
